packages/@markuplint/ml-core/src/ml-rule/types.ts

```diff
@@ -7,6 +7,8 @@ export interface MLRuleOptions<T extends RuleConfigValue, O = null> {
 	defaultLevel?: Severity;
 	defaultValue: T;
 	defaultOptions: O;
-	verify(document: Document<T, O>, translate: Translator, globalRule: RuleInfo<T, O>): Promise<Result[]>;
```

Comment:
I add the synced result into the signature for reuse the sync function in rules for `verify` and `verifySync` at the same time, and we can always `await` sync values.